lib/foreman_maintain/object_cache.rb

```diff
@@ -0,0 +1,34 @@
+require 'singleton'
+
+module ForemanMaintain
+  class ObjectCache
+    include Singleton
```

Comment:
But every run would have only one set of checks. `Procedure` are in pipeline and would come eventually, but not considered here.